server/base/src/main/java/org/apache/accumulo/server/AbstractServer.java

```diff
@@ -158,19 +160,26 @@ public String getApplicationName() {
   public abstract ServiceLock getLock();
 
   public void startServiceLockVerificationThread() {
+    Preconditions.checkState(verificationThread == null);
+    Preconditions.checkState(serverThread != null);
```

Comment:
Could add a description parameter to be used with the resulting IllegalStateException if this check fails. I'm not sure I'd know what to do with this state trace if there wasn't a description with it.